src/main/webapp/mapScript.js

```diff
@@ -45,14 +50,15 @@ function initMap() {
           map: map,
         });
 
-        let index = markersArray.length,
+        let index = globalIndex,
           stayHour = 0,
           stayMinute = 0,
-          markerName = "Place " + markersArray.length.toString();
+          markerName = "Press the settings button to choose a name";
 
         markersArray.push({
           marker: marker,
           data: {
+            index: index,
```

Comment:
Probably better to use id (or key) here rather than index, seeing that it is not really an index any more.

Reply:
Done